### DISCUSSION TITLE
NavigationView Like Win UI 3

### DISCUSSION THREAD

**Discussion by davidbuckleyni** [CommunityToolkit/Maui#935] [General]
We really need a navigation view like win ui 3 has while shell is good on mobile allot of modern apps and windows use the navigationview
 
While we could  use shell and leave it as flyout it doesnt mimic how this one works in that you can collapse it and the menu items and child items.

![image](https://user-images.githubusercontent.com/5619229/215081305-44544714-b72a-4fe1-a5bf-220df912bf6e.png)

**Comment by bijington** [comment]
It is possible to customise the [Flyout display options](https://learn.microsoft.com/en-gb/dotnet/maui/fundamentals/shell/flyout?view=net-maui-7.0#flyout-display-options) to group items together, I appreciate this might not directly match what is in the screenshot but it does show some level of customisation.

To further customise Shell you can actually put whatever you like in the Flyout through the `<Shell.FlyoutContent>` property. The following code sample shows how you can add your own `CollectionView` into the menu and customise the contents of it.

```xaml
<Shell.FlyoutContent>
    <CollectionView
        ItemsSource="{Binding Boards}"
        SelectionMode="Single"
        Margin="0,100,0,0"
        SelectedItem="{Binding CurrentBoard}">
        
        <CollectionView.ItemTemplate>
            <DataTemplate x:DataType="models:Board">
                <Label 
                    Text="{Binding Name}"
                    FontSize="20"
                    Padding="10,0,0,0" />
            </DataTemplate>
        </CollectionView.ItemTemplate>
    </CollectionView>
</Shell.FlyoutContent>
```

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
